formats/polus-tabular-to-feather-plugin/src/main.py

```diff
@@ -0,0 +1,128 @@
+                    
+            Returns:
+                Feather format file.
+                    
+            """
+        file_name = Path(file).stem
+        feather_filename = file_name + '.feather'
+        
+        logger.info('Started converting the csv file ' + file_name)
+        df = vaex.from_csv(file)
+        logger.info('writing file...')
+        os.chdir(outDir)
+        df.export_feather(feather_filename,outDir)
+
+def main(inpDir: Path,
+            outDir: Path,
+            ) -> None:
+        """ Main execution function
+        
+        All functions in your code must have docstrings attached to them, and the
+        docstrings must follow the Google Python Style:
+        https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html
+        """
+        
+        # pattern = filePattern if filePattern is not None else '.*'
+        # fp = filepattern.FilePattern(inpDir,pattern)
+        
+        
+        logger.info('outDir = {}'.format(outDir))
+        logger.info('inpDir = {}'.format(inpDir))
+        
+        # Surround with try/finally for proper error catching
+        try:
+            
+            #List the files in the directory
+            logger.info('Checking for .csv or .fcs files in the directory ')
+            fcs_filelist = list(Path(inpDir).glob('*.fcs'))
+            if not fcs_filelist:
+                raise FileNotFoundError('No .fcs files were found in the directory. Please check file directory.' )
+        
+            for each_file in fcs_filelist:
+                #Convert fcs to csv
+                logger.info("Checking for fcs files...")
+                file_ext = Path(each_file).suffix
+                if file_ext == '.fcs':
+                    logger.info('Converting fcs file to csv ')
+                    each_file = fcs_csv(each_file, inpDir)
+        finally:    
+            logger.info('Checking for .csv files in the directory ')
+            filelist = list(Path(inpDir).glob('*.csv'))
+            if not filelist:
+                raise FileNotFoundError('No .csv files were found in the directory. Please check file directory.' )
+                
+            for each_file in filelist:
+                #Read the csv file into feather
+                df_toFeather(each_file,outDir)
+                
+            logger.info("Finished all processes!")
+
+if __name__=="__main__":
+
+    ''' Argument parsing '''
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='WIPP plugin to converts Tabular Data (.FCS and .CSV) to Feather file format.')
+    
+    # Input arguments
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                        help='Input image collection to be processed by this plugin', required=False)
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
```

Comment:
We should include a filepattern input, since that is going to be a standard for all plugins.